src/resolve/resolved-varlink.c

```diff
@@ -98,6 +115,13 @@ static void vl_on_disconnect(VarlinkServer *s, Varlink *link, void *userdata) {
 
         log_debug("Client of active query vanished, aborting query.");
         dns_query_complete(q, DNS_TRANSACTION_ABORTED);
+
+        m = varlink_server_get_userdata(varlink_get_server(link));
+        if (!m)
+                return;
+
+        if (m->dns_service_subscriber)
+                m->dns_service_subscriber = mdns_service_subscriber_free(m->dns_service_subscriber);
```

Comment:
we generally, across our codebase require that destructor calls happily take NULL pointers and then become a NOP. Thus, please never do an explicit non-NULL check when calling a destructor.